### PR TITLE
Setup A/B for Demo Workflow thru Kissmetrics

### PR DESCRIPTION
**Pull request by biyeun**
In addition to switching the Schedule a Demo workflow A/B test to kissmetrics, this PR:

- adds documentation for best practice on how the a/b test framework was supposed to be used It using the `analytics_ab_test` template tag which seemed to have been forgotten about, as there seems to be an alternate method that was picked up due to sol develo. This was the original intention of the framework. 
- adds debug logging and force-refreshing variants tools to the framework to aid with debugging any issues with the setup.
- catches a condition where the `update_response` and `context` are called from two separate instances, and both generate different values. caching using the user session is now used with a fallback to the cookie to make sure that the first value read stays consistent throughout the whole experience, and no mismatches are sent back to analytics.
- renames the `AbTest` to `SessionAbTest` to differentiate it that it's for session-based caching. Ideally post-login A/B Test info should be stored in the database (similar to `Toggles`) instead of cache & cookies. Or the Toggle framework should perhaps continue to be co-opted for that.

@gbova Shouldn't affect the appcues A/B setup, just keep these implementation thoughts moving forward. 😃 
also cc @orangejenny @calellowitz 

***only for SaaS environments***

### CODE REVIEW

File: corehq/apps/analytics/static/analytix/js/hubspot.js
Comment (biyeun) [reply corehq/apps/analytics/static/analytix/js/hubspot.js]: @orangejenny can we please ignore this? lol...

File: corehq/apps/analytics/ab_tests.py
Comment (orangejenny) [inline corehq/apps/analytics/ab_tests.py]: I hear it's more typical to pass `__name__` rather than blank string. Conversation on this here: https://github.com/dimagi/commcare-hq/pull/22287#discussion_r231175201

File: corehq/apps/analytics/static/analytix/js/hubspot.js
Comment (orangejenny) [inline corehq/apps/analytics/static/analytix/js/hubspot.js]: As mentioned in other comment, it's fine to add `MutationObserver` to the "blessed" list, but I'm curious, did the other style of disabling not work?

File: corehq/apps/analytics/ab_tests.py
Comment (orangejenny) [inline corehq/apps/analytics/ab_tests.py]: Where do these messages go?

File: corehq/apps/analytics/ab_tests.py
Comment (orangejenny) [inline corehq/apps/analytics/ab_tests.py]: Thank you for adding all the docs in this file. I think it'd be helpful to make a note in https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/analytics#handling-different-environments-and-debugging that there are docs in this file relevant to debugging.

File: corehq/apps/analytics/ab_tests.py
Comment (orangejenny) [inline corehq/apps/analytics/ab_tests.py]: What does it mean to manually detect/send? I'm not sure what to avoid doing.

File: corehq/apps/analytics/static/analytix/js/drift.js
Comment (orangejenny) [inline corehq/apps/analytics/static/analytix/js/drift.js]: `shedule` => `schedule` (props for being consistent with `shedule` but still, just for everybody's sanity)

File: corehq/apps/analytics/static/analytix/js/hubspot.js
Comment (orangejenny) [inline corehq/apps/analytics/static/analytix/js/hubspot.js]: This is a requirejs module, so add this to the dependencies at the start of the file instead.

File: corehq/apps/analytics/static/analytix/js/hubspot.js
Comment (orangejenny) [inline corehq/apps/analytics/static/analytix/js/hubspot.js]: fascinating

File: corehq/apps/registration/static/registration/js/login.js
Comment (orangejenny) [inline corehq/apps/registration/static/registration/js/login.js]: Not quite following this. Is this to prevent the kiss events from firing if hubspot hasn't yet fully loaded?

**Review by orangejenny — COMMENTED** [review]
Looks good. The only point I'd like to get cleared up before merging is the comment on login.js. The hqImports in a requirejs file also need to be updated, but tests will verify that.

File: corehq/apps/analytics/ab_tests.py
Comment (biyeun) [reply corehq/apps/analytics/ab_tests.py]: in the console. intended for debugging on a dev environment

File: corehq/apps/analytics/ab_tests.py
Comment (biyeun) [reply corehq/apps/analytics/ab_tests.py]: you don't have to call `identifyTraits` also, you should pass the `context` of the `SessionAbTest` directly to the template and not modify it like it is currently. cc @gbova

File: corehq/apps/analytics/static/analytix/js/drift.js
Comment (biyeun) [reply corehq/apps/analytics/static/analytix/js/drift.js]: LOL

File: corehq/apps/analytics/static/analytix/js/hubspot.js
Comment (biyeun) [reply corehq/apps/analytics/static/analytix/js/hubspot.js]: yea the other style of disabling didn't work

File: corehq/apps/registration/static/registration/js/login.js
Comment (biyeun) [reply corehq/apps/registration/static/registration/js/login.js]: i guess this made sense when it was in `register_new_user.js` but now it's not. updated

File: corehq/apps/analytics/ab_tests.py
Comment (orangejenny) [reply corehq/apps/analytics/ab_tests.py]: aha, cool

File: corehq/apps/analytics/ab_tests.py
Comment (orangejenny) [reply corehq/apps/analytics/ab_tests.py]: Is there any code currently in prod that should be updated to reflect this? Like [new_user.ko.js](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/registration/static/registration/js/new_user.ko.js#L55) is still calling `identifyTraits`. It's not blocking this PR - and I don't think you have to do it, you can tell me or @gbova to do it - but it'd be good not to have bad examples lying around.

File: corehq/apps/analytics/static/analytix/js/hubspot.js
Comment (orangejenny) [reply corehq/apps/analytics/static/analytix/js/hubspot.js]: thanks, good to know

File: corehq/apps/analytics/ab_tests.py
Comment (biyeun) [reply corehq/apps/analytics/ab_tests.py]: will do in a separate PR since I want to make sure this PR doesn't mess up an existing test and ends up needing a revert

Comment (orangejenny) [reply corehq/apps/analytics/ab_tests.py]: good call

**Comment by biyeun** [comment]
@orangejenny finished responding to feedback

**Review by orangejenny — APPROVED** [review]

**Pull request merged**
